samples/voting/worker/Extensions.cs

```diff
@@ -10,7 +10,7 @@ public static class Extensions
     {
         public static string GetSqlConnectionString(this IConfiguration configuration)
         {
-            return configuration["connectionstring:postgres"] ?? $"Server={configuration["service:postgres:host"]};Port={configuration["service:postgres:port"]};User Id=postgres;Password=pass@word1;";
+            return configuration["connectionstrings:postgres"] ?? $"Server={configuration["service:postgres:host"]};Port={configuration["service:postgres:port"]};User Id=postgres;Password=pass@word1;";
```

Comment:
@davidfowl - this is something we should decide on...

----

Let me propose an ideal

I think the ideal thing here would be to use templating in the YAML for this.

So you'd write:

```yaml
services:
- name: postgres
  image:  postgres
  env:
    - name: POSTGRES_PASSWORD
      value: "pass@word1"
  bindings:
  - connectionString: "Server=%SERVICE__POSTGRES__HOST%;Port=%SERVICE__POSTGRES__PORT%;User Id=postgres;Password=pass@word1"
```
This doesn't require any tricky code in libraries or your app. It's roughly as simple as what this code is doing, and things like a hardcoded username/password move out of your code. It got **more** maintainable because the hardcoded password isn't used across multiple files.

You could even do better:

```yaml
services:
- name: postgres
  image:  postgres
  env:
    - name: POSTGRES_PASSWORD
      value: "pass@word1"
  bindings:
  - connectionString: "Server=%SERVICE__POSTGRES__HOST%;Port=%SERVICE__POSTGRES__PORT%;User Id=postgres;Password=%POSTGRES_PASSWORD%"
```

We could also enlighten this to use secrets.

---

I think you were onto something with the user having to do some work re: understanding connection strings, but I think it's better if it works in this direction.

Reply:
Yea, I'm a fan of this. It might be better if we have a more native syntax for the various pieces:

- Services `${service:name}`
- Env variables `${env:SERVICE__POSTGRES__HOST}`

etc.

Reply:
yeah I agree, that's nice.

Reply:
Pardon my ignorance, but are we doing anything to consolidate how environment variables value are obtained between windows and linux? Like if I have the string: `Server=%SERVICE__POSTGRES__HOST%`, how would this value be obtained on linux?